mungegithub/mungers/approval-handler.go

```diff
@@ -90,58 +97,158 @@ func (h *ApprovalHandler) Munge(obj *github.MungeObject) {
 		return
 	}
 
-	approverSet := sets.String{}
+	ownersMap := h.getApprovedOwners(files, createApproverSet(comments))
 
-	// from oldest to latest
-	for i := len(comments) - 1; i >= 0; i-- {
-		c := comments[i]
+	if err := h.updateNotification(obj, ownersMap); err != nil {
+		return
+	}
 
-		if !mungerutil.IsValidUser(c.User) {
-			continue
+	for _, approverSet := range ownersMap {
+		if approverSet.Len() == 0 {
+			if obj.HasLabel(approvedLabel) {
+				obj.RemoveLabel(approvedLabel)
+			}
+			return
 		}
+	}
 
-		fields := strings.Fields(strings.TrimSpace(*c.Body))
+	if !obj.HasLabel(approvedLabel) {
+		obj.AddLabel(approvedLabel)
+	}
+}
 
-		if len(fields) == 1 && strings.ToLower(fields[0]) == "/approve" {
-			approverSet.Insert(*c.User.Login)
-			continue
-		}
+func (h *ApprovalHandler) updateNotification(obj *github.MungeObject, ownersMap map[string]sets.String) error {
+	notificationMatcher := c.MungerNotificationName(approvalNotificationName)
+	comments, err := obj.ListComments()
+	if err != nil {
+		glog.Error("Could not list the comments for PR%v", obj.Issue.Number)
+		return err
+	}
+	notifications := c.FilterComments(comments, notificationMatcher)
+	latestNotification := notifications.GetLast()
+	if latestNotification == nil {
+		return h.createMessage(obj, ownersMap)
+	}
+	latestApprove := c.FilterComments(comments, c.CommandName(approveCommand)).GetLast()
+	if latestApprove == nil {
+		// there was already a bot notification and nothing has changed since
+		return nil
+	}
+	if latestApprove.CreatedAt.After(*latestNotification.CreatedAt) {
+		// there has been approval since last notification
+		obj.DeleteComment(latestNotification)
```

Comment:
This is different (not necessarily wrong) that how other mungers handle comment deletion. Generally they register a function which indicates if a comment is out of date and then the generic code takes care of deletion. See:

https://github.com/kubernetes/contrib/blob/master/mungegithub/mungers/stale-green-ci.go#L129

For a simple example.

You don't have to change it. We're already parsing the mess out of comments, but it is an inconsistency across mungers.